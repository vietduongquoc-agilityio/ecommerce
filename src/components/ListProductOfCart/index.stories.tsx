import { Meta, StoryObj } from "@storybook/react";
import ListProductOfCart from "./";

const meta: Meta<typeof ListProductOfCart> = {
  title: "Components/ListProductOfCart",
  component: ListProductOfCart,
};

export default meta;

type Story = StoryObj<typeof ListProductOfCart>;

export const Default: Story = {};
