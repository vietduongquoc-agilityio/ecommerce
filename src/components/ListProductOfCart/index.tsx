"use client";

import { colors, fonts } from "@/themes";
import { useRouter } from "next/navigation";
import Image from "next/image";
import living from "@/assets/Images/Living.jpg";
import deleteIcon from "@/assets/Images/DeleteIcon.png";
import Button from "../Button";

const ListProductOfCart = () => {
  const router = useRouter();
  return (
    <section
      style={{
        width: "100%",
        maxWidth: "1240px",
        margin: "0 auto",
        backgroundColor: colors.white,
        padding: "72px 0",
        display: "flex",
        justifyContent: "space-between",
      }}
    >
      <aside
        style={{
          width: "100%",
          maxWidth: "817px",
        }}
      >
        <div
          style={{
            background: colors.primary.dark,
            padding: "15px 135px",
            display: "flex",
            justifyContent: "space-between",
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
            color: colors.black,
            marginBottom: "100px",
          }}
        >
          <p style={{ width: "39.1%" }}>Product</p>
          <p style={{ width: "21.7%" }}>Price</p>
          <p style={{ width: "12.9%" }}>Quantity</p>
          <p>Subtotal</p>
        </div>
        <div
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.regular,
            display: "flex",
            width: "100%",
            maxWidth: "817px",
            justifyContent: "space-around",
          }}
        >
          <Image
            src={living}
            alt={"living"}
            style={{
              width: "108px",
              height: "105px",
              borderRadius: "10px",
            }}
          />
          <p style={{ color: colors.secondary.splight }}>Asgaard sofa</p>
          <p style={{ color: colors.secondary.splight }}>Rs. 250,000.00</p>
          <Button
            variant="outline"
            style={{
              width: "32px",
              height: "32px",
              padding: 0,
              borderRadius: "5px",
            }}
          >
            1
          </Button>
          <p>Rs. 250,000.00</p>
          <Image src={deleteIcon} alt={"deleteIcon"} />
        </div>
      </aside>
      <aside
        style={{
          width: "100%",
          maxWidth: "393px",
          background: colors.primary.dark,
          padding: "15px 75px 80px 75px",
        }}
      >
        <h2
          style={{
            fontWeight: fonts.weight.semiBold,
            fontSize: fonts.size.xl,
          }}
        >
          Cart Totals
        </h2>
        <div
          style={{
            display: "flex",
            marginTop: "61px",
            justifyContent: "space-between",
          }}
        >
          <p
            style={{
              fontWeight: fonts.weight.medium,
              fontSize: fonts.size.xs,
              color: colors.black,
            }}
          >
            Subtotal
          </p>
          <p
            style={{
              fontWeight: fonts.weight.regular,
              fontSize: fonts.size.xs,
              color: colors.secondary.splight,
            }}
          >
            Rs. 250,000.00
          </p>
        </div>
        <div
          style={{
            display: "flex",
            marginTop: "31px",
            marginBottom: "42px",
            justifyContent: "space-between",
          }}
        >
          <p
            style={{
              fontWeight: fonts.weight.medium,
              fontSize: fonts.size.xs,
              color: colors.black,
            }}
          >
            Total
          </p>
          <p
            style={{
              fontWeight: fonts.weight.medium,
              fontSize: fonts.size.md,
              color: colors.primary.yellow,
            }}
          >
            Rs. 250,000.00
          </p>
        </div>
        <Button
          variant="outline"
          style={{ padding: "14px 58px", borderRadius: fonts.borderRadius.md }}
          onClick={() => router.push("/checkout")}
        >
          Check Out
        </Button>
      </aside>
    </section>
  );
};

export default ListProductOfCart;
