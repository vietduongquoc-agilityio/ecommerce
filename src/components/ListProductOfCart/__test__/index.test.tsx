import { render } from "@testing-library/react";
import ListProductOfCart from "../";

describe("ListProductOfCart", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<ListProductOfCart />);
    expect(asFragment()).toMatchSnapshot();
  });
});
