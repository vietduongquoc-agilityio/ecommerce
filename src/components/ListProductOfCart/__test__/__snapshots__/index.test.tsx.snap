// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListProductOfCart should match snapshot 1`] = `
<DocumentFragment>
  <section
    style="width: 100%; max-width: 1240px; margin: 0px auto; background-color: rgb(255, 255, 255); padding: 72px 0px;"
  >
    <div
      style="width: 100%; max-width: 817px; background: rgb(249, 241, 231); padding: 15px 135px; display: flex; justify-content: space-between; font-size: 16px; font-weight: 500; color: rgb(0, 0, 0); margin-bottom: 100px;"
    >
      <p
        style="width: 39.1%;"
      >
        Product
      </p>
      <p
        style="width: 21.7%;"
      >
        Price
      </p>
      <p
        style="width: 12.9%;"
      >
        Quantity
      </p>
      <p>
        Subtotal
      </p>
    </div>
    <div
      style="font-size: 16px; font-weight: 400; display: flex; width: 100%; max-width: 817px; justify-content: space-around;"
    >
      <img
        alt="living"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent; width: 108px; height: 105px; border-radius: 10px;"
        width="40"
      />
      <p
        style="color: rgb(159, 159, 159);"
      >
        Asgaard sofa
      </p>
      <p
        style="color: rgb(159, 159, 159);"
      >
        Rs. 250,000.00
      </p>
      <button
        style="padding: 0px; font-size: 18px; font-weight: 600; text-align: center; width: 32px; height: 32px; cursor: pointer; transition: all 0.3s ease; border: 1px solid #000000; background-color: transparent; color: rgb(0, 0, 0); border-radius: 5px;"
      >
        1
      </button>
      <p>
        Rs. 250,000.00
      </p>
      <img
        alt="deleteIcon"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
  </section>
</DocumentFragment>
`;
